crates/cast/src/starknet_commands/account/add.rs

```diff
@@ -0,0 +1,101 @@
+use crate::starknet_commands::account::{prepare_account_json, write_account_to_accounts_file};
+use anyhow::{ensure, Result};
+use camino::Utf8PathBuf;
+use cast::get_chain_id;
+use cast::helpers::response_structs::AccountAddResponse;
+use cast::helpers::scarb_utils::CastConfig;
+use clap::Args;
+use starknet::core::types::BlockTag::Latest;
+use starknet::core::types::{BlockId, FieldElement};
+use starknet::providers::{
+    jsonrpc::{HttpTransport, JsonRpcClient},
+    Provider,
+};
+use starknet::signers::SigningKey;
+
+#[derive(Args, Debug)]
+#[command(about = "Add an account to the accounts file")]
+pub struct Add {
+    /// Name of the account to be added
+    #[clap(short, long)]
+    pub name: String,
+
+    /// Address of the account
+    #[clap(short, long)]
+    pub address: FieldElement,
+
+    /// Class hash of the account
+    #[clap(short, long)]
+    pub class_hash: Option<FieldElement>,
+
+    /// Account deployment status
+    /// If not passed, sncast will check whether the account is deployed or not
+    #[clap(short, long)]
+    pub deployed: bool,
+
+    /// Account private key
+    #[clap(long)]
+    pub private_key: FieldElement,
+
+    /// Account public key
+    #[clap(long)]
+    pub public_key: Option<FieldElement>,
+
+    /// Salt for the address
+    #[clap(short, long)]
+    pub salt: Option<FieldElement>,
+
+    /// If passed, a profile with corresponding data will be created in Scarb.toml
+    #[clap(long)]
+    pub add_profile: bool,
+}
+
+pub async fn add(
+    config: &CastConfig,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
+    provider: &JsonRpcClient<HttpTransport>,
+    add: &Add,
+) -> Result<AccountAddResponse> {
+    let private_key = &SigningKey::from_secret_scalar(add.private_key);
+    if let Some(public_key) = &add.public_key {
+        ensure!(
+            public_key == &private_key.verifying_key().scalar(),
+            "public key mismatch"
+        );
+    }
+
+    let deployed = if add.deployed {
+        true
+    } else if provider
+        .get_class_hash_at(BlockId::Tag(Latest), add.address)
+        .await
+        .is_ok()
```

Comment:
can this potentially break cast somehow? I mean some random problems with rpc node, or something

Reply:
What exactly do you mean by breaking cast? If the provider request fails for some reason, deployed will be set to false

Reply:
will it panic? we should avoid that

Reply:
It won't, `await` returns Result, and here we just check if the Result is Ok